File: 541-reverse-string-ii/reverse-string-ii.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        r = ''
        i = 0
        incr = k*2
        while i<len(s):
            curr = s[i:i+k] 
            r += curr[::-1] + s[i+k:i+incr]
            i+=incr
        return r

File: 541-reverse-string-ii/reverse-string-ii.java
class Solution {
    public String reverseStr(String s, int k) {
        if(s.length() == k){
            return new StringBuilder(s).reverse().toString();
        }
        String revstr = "";
        int i = 0;
        int n = s.length();
        while(i<s.length()){
            int endfirst = Math.min(i+k,n);
            int endsec = Math.min(i+k*2,n);
            revstr += new StringBuilder(s.substring(i,endfirst)).reverse().toString() + s.substring(endfirst,endsec);
            i = i+2*k;
        }
        return revstr;
    }
}